src/components/address-book/ImportDialog/validation.ts

```diff
@@ -0,0 +1,74 @@
+import type { ParseResult } from 'papaparse'
+
+import { validateAddress, validateChainId } from '@/utils/validation'
+
+export const abCsvReaderValidator = ({ type, size }: File): string[] | undefined => {
+  if (type !== 'text/csv') {
+    return ['Address book must be a CSV file']
+  }
+
+  if (size > 1_000_000) {
+    return ['Address book cannot be larger than 1MB']
+  }
+}
+
+export const hasValidAbHeader = (header: string[]) => {
+  return header.length === 3 && header[0] === 'address' && header[1] === 'name' && header[2] === 'chainId'
+}
+
+export const hasCompleteAbEntries = (entries: string[][]) => {
+  return entries.every((entry) => entry.every(Boolean))
+}
+
+export const hasValidAbEntryAddresses = (entries: string[][]) => {
+  return entries.every((entry) => entry.length >= 1 && !validateAddress(entry[0]))
+}
+
+export const hasValidAbEntryChainIds = (entries: string[][]) => {
+  return entries.every((entry) => {
+    if (entry.length < 3) {
+      return false
+    }
+    const chainId = entry[2].replace(/\s/g, '')
+    return !validateChainId(chainId)
+  })
+}
+
+export const abOnUploadValidator = ({ data, errors }: ParseResult<string[]>): string | undefined => {
+  const [header, ...entries] = data
+
+  // papaparse error
+  if (errors.length > 0) {
+    return errors[0].message
+  }
+
+  // Empty CSV
+  if (data.length === 0) {
+    return 'CSV file is empty'
+  }
+
+  // Wrong header
+  if (!hasValidAbHeader(header)) {
+    return 'Invalid or corrupt address book'
+  }
+
+  // No entries
+  if (entries.length === 0) {
+    return 'No entries found in address book'
+  }
+
+  // An entry has empty value
+  if (!hasCompleteAbEntries(entries)) {
+    return 'Address book contains entries with empty fields'
+  }
```

Comment:
Do we need this validation? Currently, we just import everything that can be imported and ignore the rest which is a very simple solution.

Reply:
That's an elegant solution that I didn't think of. I'll filter out the empty fields then.

Reply:
Good catch. I've adjusted it to ignore them.